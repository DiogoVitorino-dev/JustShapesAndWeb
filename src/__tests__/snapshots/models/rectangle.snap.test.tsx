import { render, renderHook } from "@testing-library/react-native";
import { useSharedValue } from "react-native-reanimated";

import Rectangle, { RectangleData } from "@/models/geometric/rectangle";

describe("Rectangle model - Snapshot test", () => {
  it("Should renders correctly", () => {
    const data = renderHook(() =>
      useSharedValue<RectangleData>({ x: 25, y: 50 }),
    );

    const tree = render(<Rectangle data={data.result.current} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
