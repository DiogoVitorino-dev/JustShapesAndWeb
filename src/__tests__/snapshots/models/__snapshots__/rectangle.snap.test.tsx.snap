// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rectangle model - Snapshot test Should renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "height": 100,
        "left": 25,
        "top": 50,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 50,
      },
    }
  }
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(255, 64, 20)",
      "height": 100,
      "left": 25,
      "position": "absolute",
      "top": 50,
      "transform": [
        {
          "rotate": "0deg",
        },
      ],
      "width": 50,
    }
  }
  testID="rectangleModel"
/>
`;
